tests/e2e/testsuites/testsuites.go

```diff
@@ -71,6 +75,75 @@ func (t *TestStorageClass) Cleanup() {
 	framework.ExpectNoError(err)
 }
 
+type TestVolumeSnapshotClass struct {
+	client              restclientset.Interface
+	volumeSnapshotClass *v1alpha1.VolumeSnapshotClass
+	namespace           *v1.Namespace
+}
+
+func NewTestVolumeSnapshotClass(c restclientset.Interface, ns *v1.Namespace, vsc *v1alpha1.VolumeSnapshotClass) *TestVolumeSnapshotClass {
+	return &TestVolumeSnapshotClass{
+		client:              c,
+		volumeSnapshotClass: vsc,
+		namespace:           ns,
+	}
+}
+
+func (t *TestVolumeSnapshotClass) Create() {
+	By("creating a VolumeSnapshotClass")
+	var err error
+	t.volumeSnapshotClass, err = snapshotclientset.New(t.client).VolumesnapshotV1alpha1().VolumeSnapshotClasses().Create(t.volumeSnapshotClass)
+	framework.ExpectNoError(err)
+}
+
+func (t *TestVolumeSnapshotClass) CreateSnapshot(pvc *v1.PersistentVolumeClaim) *v1alpha1.VolumeSnapshot {
+	By("creating a VolumeSnapshot for " + pvc.Name)
+	snapshot := &v1alpha1.VolumeSnapshot{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       VolumeSnapshotKind,
+			APIVersion: SnapshotAPIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "volume-snapshot-",
+			Namespace:    t.namespace.Name,
+		},
+		Spec: v1alpha1.VolumeSnapshotSpec{
+			VolumeSnapshotClassName: &t.volumeSnapshotClass.Name,
+			Source: &v1.TypedLocalObjectReference{
+				Kind: "PersistentVolumeClaim",
+				Name: pvc.Name,
+			},
+		},
+	}
+	snapshot, err := snapshotclientset.New(t.client).VolumesnapshotV1alpha1().VolumeSnapshots(t.namespace.Name).Create(snapshot)
+	framework.ExpectNoError(err)
+	return snapshot
+}
+
+func (t *TestVolumeSnapshotClass) ReadyToUse(snapshot *v1alpha1.VolumeSnapshot) {
+	By("waiting for VolumeSnapshot to be ready to use - " + snapshot.Name)
+	err := wait.Poll(15*time.Second, 5*time.Minute, func() (bool, error) {
+		vs, err := snapshotclientset.New(t.client).VolumesnapshotV1alpha1().VolumeSnapshots(t.namespace.Name).Get(snapshot.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("did not see ReadyToUse: %v", err)
+		}
+		return vs.Status.ReadyToUse, nil
+	})
+	framework.ExpectNoError(err)
+}
+
+func (t *TestVolumeSnapshotClass) DeleteSnapshot(vs *v1alpha1.VolumeSnapshot) {
+	By("deleting a VolumeSnapshot " + vs.Name)
+	err := snapshotclientset.New(t.client).VolumesnapshotV1alpha1().VolumeSnapshots(t.namespace.Name).Delete(vs.Name, &metav1.DeleteOptions{})
+	framework.ExpectNoError(err)
+}
+
+func (t *TestVolumeSnapshotClass) Cleanup() {
+	framework.Logf("deleting VolumeSnapshotClass %s", t.volumeSnapshotClass.Name)
+	err := snapshotclientset.New(t.client).VolumesnapshotV1alpha1().VolumeSnapshotClasses().Delete(t.volumeSnapshotClass.Name, nil)
```

Comment:
I feel we have too many clients here, and it sometimes is confusing. Could we create the snapshotclientset at the very beginning inside `BeforeEach` instead of creating restclient there and passing it all the way down here? How do you think?

Reply:
After putting it in its own `Describe` it feels cleaner since all of these tests would require the client.